src/parser/view_transform.c

```diff
@@ -13989,36 +14019,76 @@ mq_update_analytic_sort_spec_expr (PARSER_CONTEXT * parser, PT_NODE * spec, PT_N
 
 	  for (order = order_list; order; order = order->next)
 	    {
-	      PT_NODE *old_attr, *new_attr;
-	      int index;
+	      PT_NODE *old_select_node, *new_select_node;
+	      PT_NODE *from;
+	      int old_index, index;
 
 	      if (!PT_IS_VALUE_NODE (order->info.sort_spec.expr))
 		{
 		  continue;
 		}
 
 	      value = order->info.sort_spec.expr;
-	      /* retrieve the order-th node from the old_attrs */
-	      old_attr = pt_resolve_sort_spec_expr (parser, order, old_attrs);
-	      if (old_attr == NULL)
+	      old_index = value->info.value.data_value.i;
+
+	      if (old_select_list->type_enum != PT_TYPE_OBJECT)
 		{
-		  assert (false);
+		  old_index -= 1;
+		}
+
+	      old_select_node = pt_get_node_from_list (old_select_list, old_index);
+	      if (old_select_node == NULL)
+		{
+		  PT_ERRORmf (parser, order, MSGCAT_SET_PARSER_SEMANTIC, MSGCAT_SEMANTIC_SORT_SPEC_RANGE_ERR,
```

Comment:
```suggestion
		  PT_ERRORmf (parser, new_query, MSGCAT_SET_PARSER_SEMANTIC, MSGCAT_SEMANTIC_SORT_SPEC_RANGE_ERR,
```
에러 설정은 상위 노드에 하는 것이 맞는 것 같습니다.

Reply:
확인했습니다.